core/src/main/scala/org/http4s/headers/X-Forwarded-For.scala

```diff
@@ -18,13 +18,20 @@ package org.http4s
 package headers
 
 import cats.data.NonEmptyList
+import cats.parse._
 import java.net.InetAddress
-import org.http4s.parser.HttpHeaderParser
+import org.http4s.internal.parsing.Rfc7230
 import org.http4s.util.Writer
 
 object `X-Forwarded-For` extends HeaderKey.Internal[`X-Forwarded-For`] with HeaderKey.Recurring {
   override def parse(s: String): ParseResult[`X-Forwarded-For`] =
-    HttpHeaderParser.X_FORWARDED_FOR(s)
+    ParseResult.fromParser(parser, "Invalid X-Forwarded-For")(s)
+  private[http4s] val parser: Parser[`X-Forwarded-For`] =
+    Rfc7230
+      .headerRep1(
+        (Uri.Ipv4Address.parser.map(_.toInet4Address).backtrack | Uri.Ipv6Address.parser.map(
```

Comment:
This one can't be a hostname?

Reply:
it's not defined in any spec but according to the mdn it's just ip addresses: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For